src/bind-all.ts

```diff
@@ -35,254 +35,31 @@ function getBinding(original: Binding, sharedOptions?: boolean | AddEventListene
   return binding;
 }
 
-export function bindAll<Target extends EventTarget, Type extends string>(
-  target: Target,
-  bindings: [Binding<Target, Type>],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<Target extends EventTarget, Type1 extends string, Type2 extends string>(
-  target: Target,
-  bindings: [Binding<Target, Type1>, Binding<Target, Type2>],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<
-  Target extends EventTarget,
-  Type1 extends string,
-  Type2 extends string,
-  Type3 extends string,
->(
-  target: Target,
-  bindings: [Binding<Target, Type1>, Binding<Target, Type2>, Binding<Target, Type3>],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<
-  Target extends EventTarget,
-  Type1 extends string,
-  Type2 extends string,
-  Type3 extends string,
-  Type4 extends string,
->(
-  target: Target,
-  bindings: [
-    Binding<Target, Type1>,
-    Binding<Target, Type2>,
-    Binding<Target, Type3>,
-    Binding<Target, Type4>,
-  ],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<
-  Target extends EventTarget,
-  Type1 extends string,
-  Type2 extends string,
-  Type3 extends string,
-  Type4 extends string,
-  Type5 extends string,
->(
-  target: Target,
-  bindings: [
-    Binding<Target, Type1>,
-    Binding<Target, Type2>,
-    Binding<Target, Type3>,
-    Binding<Target, Type4>,
-    Binding<Target, Type5>,
-  ],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
-export function bindAll<
-  Target extends EventTarget,
-  Type1 extends string,
-  Type2 extends string,
-  Type3 extends string,
-  Type4 extends string,
-  Type5 extends string,
-  Type6 extends string,
->(
-  target: Target,
-  bindings: [
-    Binding<Target, Type1>,
-    Binding<Target, Type2>,
-    Binding<Target, Type3>,
-    Binding<Target, Type4>,
-    Binding<Target, Type5>,
-    Binding<Target, Type6>,
-  ],
-  sharedOptions?: boolean | AddEventListenerOptions,
-): UnbindFn;
 export function bindAll<
```

Comment:
To confirm my understanding, the signature of `bindAll` has changed:

```diff
- bindAll<{} as HTMLElement, 'click', 'mousedown'>
+ bindAll<{} as HTMLElement, ['click', 'mousedown']>
```

But if you are using inference then nothing has changed. Is that right?

Reply:
Yes, that's the hope at least.